src/tests/unit/tests/electron/flux/action-creator/window-state-action-creator.test.ts

```diff
@@ -79,4 +93,87 @@ describe(WindowStateActionCreator, () => {
         setWindowStatePayload.verifyAll();
         windowFrameActionCreatorMock.verifyAll();
     });
+
+    describe('calling setRoute with view other than deviceConnectView', () => {
+        const testPayload: RoutePayload = {
+            routeId: 'resultsView',
+        };
+
+        let setRouteActionMock;
+
+        beforeEach(() => {
+            setRouteActionMock = Mock.ofType<Action<RoutePayload>>();
+            setRouteActionMock.setup(s => s.invoke(testPayload)).verifiable(Times.once());
+            windowStateActionsMock
+                .setup(actions => actions.setRoute)
+                .returns(() => setRouteActionMock.object)
+                .verifiable(Times.once());
+        });
+
+        afterEach(() => {
+            setRouteActionMock.verifyAll();
+            windowFrameActionCreatorMock.verifyAll();
```

Comment:
Should we put these verify statements in each of the test cases? To my understanding, the afterEach block is more for cleanup

Reply:
Failures in afterEach will still cause the specific test to fail, so there's no functional difference. I prefer to group the common code into common places, so that the specific unit tests are short and targeted to the exact scenario. I'll put a post in the chat to see if anyone has really strong opinions either way.

Reply:
Moved them into setRouteNonDeviceViewCore, which is already the helper method where all code converges